src/rules.ts

```diff
@@ -254,21 +234,61 @@ const isMatch: Matcher = (rule, options) => {
   return matches.length ? matches.every((match) => match === true) : false;
 };
 
-export const getMatchingRules = (
-  rules: Rule[],
-  files: RuleFile[],
-  event: Event,
-  patchContent: string[] = []
-): MatchingRule[] => {
-  const fileNames = files.map(({ filename }) => filename);
-
-  const matchingRules = rules.reduce((memo, rule) => {
-    if (isMatch(rule, { event, patch: patchContent, fileNames })) {
-      return [...memo, { ...rule, matched: true }];
-    } else {
-      return memo;
-    }
-  }, [] as MatchingRule[]);
+export class Rules extends Array<Rule> {
+  public constructor(...items: Rule[]) {
+    super(...items);
+  }
+  static loadFromLocation(location: string): Rules {
```

Comment:
What do you think about putting the logic in this function into the constructor, instead of having this static method which returns an instance of `Rules`?

Then we could create a new `Rules` instance by passing in only a location, i.e. `new Rules('./path/to/rules/')`

Reply:
I have a couple of ways of answering why I made that choice, but I'm gonna go with the most important ones. 

1) Extending Array Class to simplify use: The idea of extending the Array class is to use the Array-like capabilities to simplify the interaction with the collection of rules. Changing the constructor will break all the array behavior, given that map (for example) creates arrays as it iterates using. internally the constructor. 

2) Separation of concerns: `loadingFromLocation` Interacts with the fileSystem. and that behavior will change over time as we evolve our code ( ex: new YAML rules we (you) are gonna introduce) so coupling both will prove difficult in the future to test, given that we will have to mock it in 90% of the time. 

3) Idiomatic: as you know, classes are different from functions, particularly in this case I'm using them as a way to standardized the API we are gonna, later on, interact with. with that in mind, `loadFromLocation` could evolve or we could add `loadFromYamlLocation` (for instance) and avoid having to confusingly extend the constructor to understand what should be loaded and from where. instead, we can extend the API (if necessary), to accommodate new ways to populate the rules.  

in a personal note: I like the static method because it simplifies the way we can populate the Rules. Giving us constructor like behavior without breaking the contract with Array class.  (best of both worlds)